src/main/utils/mavenUtils.ts

```diff
@@ -0,0 +1,319 @@
+                currNode.pomPath = path.dirname(pom.fsPath);
+                currNode.rawDependencies = rawDependencies;
+                currNode.parentId = parentId;
+                addNode(result, currNode);
+            }
+        } catch (error) {
+            treesManager.logManager.logError(error, false);
+            treesManager.logManager.logMessage(
+                'Could not get dependencies tree from pom.xml (path: ' +
+                    pom +
+                    ').\n' +
+                    'Possible cause: The project needs to be installed by maven. Install it by running "mvn clean install" from ' +
+                    pom +
+                    '.',
+                'ERR'
+            );
+        }
+    });
+    return result;
+}
+// If the node have parent do:
+// 1. check if the parent already in the tree,
+//     1.1 if its add to its child
+//     1.2 else create the parent and add the parent to the array
+// 2.otherwise add the node to the array
+export function addNode(array: PomTree[], node: PomTree) {
+    if (!!node.parentId) {
+        const parentNode: PomTree | undefined = deepSearchNode(array, node.parentId);
+        if (!!parentNode) {
+            parentNode.addChild(node);
+        } else {
+            const ParentPom: PomTree = new PomTree(node.parentId);
+            ParentPom.addChild(node);
+            array.push(ParentPom);
+        }
+    } else {
+        array.push(node);
+    }
+}
+
+export function verifyMavenInstalled(): boolean {
+    try {
+        exec.execSync('mvn -version');
+    } catch (error) {
+        return false;
+    }
+    return true;
+}
+
+export function getParentInfo(pomXml: vscode.Uri): string {
+    const parentText: RegExpMatchArray | null = fs
+        .readFileSync(pomXml.fsPath)
+        .toString()
+        .match(/<parent>(.|\s)*?<\/parent>/);
+    if (parentText && parentText[0]) {
+        const groupId: RegExpMatchArray | null = parentText[0].toString().match(/(?<=\<groupId>).+?(?=\<\/groupId>)/i);
+        const artifactId: RegExpMatchArray | null = parentText[0].toString().match(/(?<=\<artifactId>).+?(?=\<\/artifactId>)/i);
+        const version: RegExpMatchArray | null = parentText[0].toString().match(/(?<=\<version>).+?(?=\<\/version>)/i);
+        return groupId + ':' + artifactId + ':' + version;
+    }
+    return '';
+}
+
+export function deepSearchNode(array: PomTree[], pomId: string): PomTree | undefined {
+    for (let index: number = 0; index < array.length; index++) {
+        if (pomId === array[index].pomId) {
+            return array[index];
+        }
+        if (!!array[index].children) {
+            deepSearchNode(array[index].children!, pomId);
+        }
+    }
+    return;
+}
+
+export function searchPomId(array: PomTree[], pomId: string): number {
+    for (let index: number = 0; index < array.length; index++) {
+        if (pomId === array[index].pomId) {
+            return index;
+        }
+    }
+    return -1;
+}
```

Comment:
Can be one-line inside PomTree:
```typescript
public searchPomId(pomTrees: PomTree[], pomId: string): number {
    return pomTrees.findIndex(pomTree => pomTree.pomId === pomId);
}
```
Or even better if you make it a PomTree method and search locally under this